packages/react-components/src/components/ParticipantList.tsx

```diff
@@ -197,34 +197,47 @@ export const ParticipantList = (props: ParticipantListProps): JSX.Element => {
     return onRenderParticipant ? participants : getParticipantsForDefaultRender(participants, excludeMe, myUserId);
   }, [participants, excludeMe, myUserId, onRenderParticipant]);
 
-  const createParticipantMenuItems = (participant: ParticipantListParticipant): IContextualMenuItem[] => {
-    let menuItems: IContextualMenuItem[] = [];
-    let disabled = !participant.isRemovable;
-    /* @conditional-compile-remove(rooms) */
-    const isRemovable = _usePermissions().removeParticipantButton;
-    /* @conditional-compile-remove(rooms) */
-    disabled = !isRemovable || disabled;
-    if (participant.userId !== myUserId && onRemoveParticipant) {
-      menuItems.push({
-        key: 'remove',
-        text: strings.removeButtonLabel,
-        onClick: () => onRemoveParticipant(participant.userId),
-        itemProps: {
-          styles: props.styles?.participantItemStyles?.participantSubMenuItemsStyles
-        },
-        disabled: disabled,
-        'data-ui-id': ids.participantListRemoveParticipantButton
-      });
-    }
+  const createParticipantMenuItems = useCallback(
+    (participant: ParticipantListParticipant): IContextualMenuItem[] => {
+      let menuItems: IContextualMenuItem[] = [];
+      let disabled = !participant.isRemovable;
+      /* @conditional-compile-remove(rooms) */
+      const isRemovable = _usePermissions().removeParticipantButton;
+      /* @conditional-compile-remove(rooms) */
+      disabled = !isRemovable || disabled;
+      if (participant.userId !== myUserId && onRemoveParticipant) {
+        menuItems.push({
+          key: 'remove',
+          text: strings.removeButtonLabel,
+          onClick: () => onRemoveParticipant(participant.userId),
+          itemProps: {
+            styles: props.styles?.participantItemStyles?.participantSubMenuItemsStyles
+          },
+          disabled: disabled,
+          'data-ui-id': ids.participantListRemoveParticipantButton
+        });
+      }
 
-    if (onFetchParticipantMenuItems) {
-      menuItems = onFetchParticipantMenuItems(participant.userId, myUserId, menuItems);
-    }
+      if (onFetchParticipantMenuItems) {
+        menuItems = [...onFetchParticipantMenuItems(participant.userId, myUserId, menuItems)];
```

Comment:
Why did we enforce the type of menu items to array, is it not array already?

Reply:
Oops great catch! That was from me testing, I will revert this line!

Reply:
reverted